swig/include/python/ogr_python.i

```diff
@@ -643,8 +643,52 @@ def ReleaseResultSet(self, sql_lyr):
         return self.CreateFieldFromArrowSchema(schema, options)
 
 
+    def WriteArrow(self, obj, requested_schema=None, createFieldsFromSchema=None, options=[]):
+        """Write the content of the passed object, which must implement the
+           __arrow_c_stream__ or __arrow_c_array__ interface, into the layer.
+
+           Parameters
+           ----------
+           obj:
+               Object implementing the __arrow_c_stream__ or __arrow_c_array__ interface
+
+           requested_schema: PyCapsule, default None
+               The schema to which the stream should be casted, passed as a
+               PyCapsule containing a C ArrowSchema representation of the
+               requested schema.
```

Comment:
I don't think there is already much experience with this `requested_schema` keyword, but you could also accept any "schema-like" object that implements `__arrow_c_schema__` (and call that method to get the capsule before passing it to `__arrow_c_array/stream__`). 
That makes it similar to the main `obj` keyword, where the user can also pass any object implementing the protocol, instead of passing a capsule itself. 

cc @kylebarron

Reply:
The implementation just passes requested_schema to ``obj.__arrow_c_stream__(requested_schema=requested_schema)`` . So if ``__arrow_c_stream__()`` also accepts an object that implements ``__arrow_c_schema__`` that will work. I just copied the official documentation from https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#arrowstream-export

Reply:
Yes, the dunder methods expect a capsule for the `requested_schema`, that's indeed how it is suppose to work (and is documented). 

But here the user is not calling that directly, they are using a public user-facing ``WriteArrow`` method, and so there you can do whatever you want. And what I did for other projects was to accept any schema-like object.

What I meant is essentially just adding this:

```
if hasattr(requested_schema, "__arrow_c_schema__"):
    requested_schema = requested_schema.__arrow_c_schema__()
```

before passing `requested_schema` to `obj.__arrow_c_stream__(requested_schema=requested_schema)`

Reply:
ok, got it. Commit implementing that added

Reply:
Looks good!